hasura/metadata/databases/default/tables/public_action.yaml

```diff
@@ -51,11 +51,15 @@ insert_permissions:
   - role: user
     permission:
       check:
-        app:
-          team:
-            users:
-              id:
-                _eq: X-Hasura-User-Id
+        _and:
+          - app:
+              team:
+                roles:
+                  user_id:
+                    _eq: X-Hasura-User-Id
+          - app:
+              team_id:
+                _eq: X-Hasura-Team-Id
```

Comment:
how are we planning to do the selection of which team a user is interacting with. I'm partial to doing something that relies on the URL (e.g. https://developer.worldcoin.org/team/team_123/apps), that way when you share URLs they'll work reliably. 

if we follow that approach, then a single JWT will allow manipulating any team to which the user has permissions and the query needs to be constrained, i.e. drop this restriction